### PR TITLE
refactor: migrate base components to RTL part 1

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Migrated the following test components to RTL:
  - `app/base/components/ActionForm/ActionForm.test.tsx`
  - `app/base/components/ArchitectureSelect/ArchitectureSelect.test.tsx`
  - `app/base/components/ArrowPagination/ArrowPagination.test.tsx`
  - `app/base/components/CertificateDetails/CertificateDetails.test.tsx`
  - `app/base/components/CertificateDownload/CertificateDownload.test.tsx`
  - `app/base/components/CertificateFields/CertificateFields.test.tsx`
  - `app/base/components/CertificateMetadata/CertificateMetadata.test.tsx`
  - `app/base/components/ColumnToggle/ColumnToggle.test.tsx`
  - `app/base/components/CopyButton/CopyButton.test.tsx`

### CODE REVIEW

File: src/app/base/components/CertificateFields/CertificateFields.test.tsx [hunk 1]
```diff
@@ -16,12 +16,17 @@ describe("CertificateFields", () => {
         />
       </Formik>
     );
-    expect(wrapper.find("textarea[name='certificate']").exists()).toBe(false);
-    expect(wrapper.find("textarea[name='key']").exists()).toBe(false);
+
+    expect(
+      screen.queryByRole("textbox", { name: "Upload certificate" })
```
Comment (huwshimi) [inline src/app/base/components/CertificateFields/CertificateFields.test.tsx:21]: Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.Could use an enum here too.

I think my keyboard got sick of my pasting this everywhere so it just pasted it a million times for some reason.

File: src/app/base/components/CertificateFields/CertificateFields.test.tsx [hunk 2]
```diff
@@ -32,12 +37,17 @@ describe("CertificateFields", () => {
         />
       </Formik>
     );
-    expect(wrapper.find("textarea[name='certificate']").exists()).toBe(true);
-    expect(wrapper.find("textarea[name='key']").exists()).toBe(true);
+
+    expect(
+      screen.getByRole("textbox", { name: "Upload certificate" })
```
Comment (huwshimi) [inline src/app/base/components/CertificateFields/CertificateFields.test.tsx:42]: Could use an enum here too.

File: src/app/base/components/CertificateFields/CertificateFields.test.tsx [hunk 3]
```diff
@@ -32,12 +37,17 @@ describe("CertificateFields", () => {
...
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole("textbox", { name: "Upload private key" })
```
Comment (huwshimi) [inline src/app/base/components/CertificateFields/CertificateFields.test.tsx:45]: Could use an enum here too.

File: src/app/base/components/CertificateFields/CertificateFields.test.tsx [hunk 4]
```diff
@@ -49,8 +59,9 @@ describe("CertificateFields", () => {
         />
       </Formik>
     );
-    expect(wrapper.find("textarea[name='custom-private-key']").exists()).toBe(
-      true
-    );
+
+    expect(
+      screen.getByRole("textbox", { name: "Upload private key" })
```
Comment (huwshimi) [inline src/app/base/components/CertificateFields/CertificateFields.test.tsx:64]: Could use an enum here too.

File: src/app/base/components/CertificateMetadata/__snapshots__/CertificateMetadata.test.tsx.snap [hunk 5]
```diff
@@ -1,27 +1,60 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`CertificateMetadata renders 1`] = `
```
Comment (huwshimi) [inline src/app/base/components/CertificateMetadata/__snapshots__/CertificateMetadata.test.tsx.snap:3]: Another snapshot we could potentially remove.

File: src/app/base/components/ColumnToggle/__snapshots__/ColumnToggle.test.tsx.snap [hunk 6]
```diff
@@ -1,19 +1,18 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`ColumnToggle  renders 1`] = `
```
Comment (huwshimi) [inline src/app/base/components/ColumnToggle/__snapshots__/ColumnToggle.test.tsx.snap:3]: And another snapshot.

File: src/app/base/components/CopyButton/__snapshots__/CopyButton.test.tsx.snap [hunk 7]
```diff
@@ -1,29 +1,19 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`CopyButton can render 1`] = `
```
Comment (huwshimi) [inline src/app/base/components/CopyButton/__snapshots__/CopyButton.test.tsx.snap:3]: One more snapshot.

**Review by huwshimi — COMMENTED** [review]
A few comments for your consideration.

File: src/app/base/components/ActionForm/ActionForm.test.tsx [hunk 8]
```diff
@@ -38,12 +38,14 @@ describe("ActionForm", () => {
       </Provider>
     );
 
-    expect(wrapper.find("Spinner").exists()).toBe(true);
+    expect(
+      screen.getByRole("alert", { name: /Loading form/ })
```
[the remark this answers is not in the document]
Comment (Caleb-Ellis) [reply src/app/base/components/ActionForm/ActionForm.test.tsx:42]: Right, I was trying to make as little changes to the component files as possible (like introducing enums) to simplify these PRs, but I suppose this is the best opportunity to actually improve the tests.

File: src/app/base/components/ArchitectureSelect/__snapshots__/ArchitectureSelect.test.tsx.snap [hunk 9]
```diff
@@ -2,39 +2,29 @@
 
 exports[`ArchitectureSelect renders a list of all architectures in state 1`] = `
 <select
```
[the remark this answers is not in the document]
Comment (Caleb-Ellis) [reply src/app/base/components/ArchitectureSelect/__snapshots__/ArchitectureSelect.test.tsx.snap:4]: I'm all for it

**Review by huwshimi — APPROVED** [review]

**Pull request merged**